logsmanagement/functions.c

```diff
@@ -0,0 +1,1119 @@
+                                                    FACET_KEY_OPTION_FTS);
+
+    facets_register_key_name(facets, "log_type",    FACET_KEY_OPTION_FACET | 
+                                                    FACET_KEY_OPTION_FTS);
+
+    facets_register_key_name(facets, "filename",    FACET_KEY_OPTION_FACET | 
+                                                    FACET_KEY_OPTION_FTS);
+
+    facets_register_key_name(facets, "basename",    FACET_KEY_OPTION_FACET | 
+                                                    FACET_KEY_OPTION_FTS);
+
+    facets_register_key_name(facets, "chartname",   FACET_KEY_OPTION_VISIBLE | 
+                                                    FACET_KEY_OPTION_FACET | 
+                                                    FACET_KEY_OPTION_FTS);
+
+    facets_register_key_name(facets, "message",     FACET_KEY_OPTION_NEVER_FACET | 
+                                                    FACET_KEY_OPTION_MAIN_TEXT | 
+                                                    FACET_KEY_OPTION_VISIBLE |
+                                                    FACET_KEY_OPTION_FTS);
+
+    bool    info        = false, 
+            data_only   = false, 
+            progress    = false, 
+            /* slice       = true, */
+            delta       = false, 
+            tail        = false;
+    time_t after_s = 0, before_s = 0;
+    usec_t anchor = 0;
+    usec_t if_modified_since = 0;
+    size_t last = 0;
+    FACETS_ANCHOR_DIRECTION direction = FACETS_ANCHOR_DIRECTION_BACKWARD;
+    const char *query = NULL;
+    const char *chart = NULL;
+    const char *source = NULL;
+    const char *progress_id = NULL;
+    // SD_JOURNAL_FILE_SOURCE_TYPE source_type = SDJF_ALL;
+    // size_t filters = 0;  
+
+    buffer_json_member_add_object(wb, "_request");
+
+    logs_query_params_t query_params = {0};
+    unsigned long req_quota = 0;
+
+    unsigned int fn_off = 0, cn_off = 0;
+
+    char *words[LOGS_MANAG_MAX_PARAMS] = { NULL };
+    size_t num_words = quoted_strings_splitter_pluginsd(function, words, LOGS_MANAG_MAX_PARAMS);
+    for(int i = 1; i < LOGS_MANAG_MAX_PARAMS ; i++) {
+        char *keyword = get_word(words, num_words, i);
+        if(!keyword) break;
+
+        
+        if(!strcmp(keyword, LOGS_MANAG_FUNC_PARAM_HELP)){
+            BUFFER *wb = buffer_create(0, NULL);
+            buffer_sprintf(wb, FUNCTION_LOGSMANAGEMENT_HELP_LONG);
+            netdata_mutex_lock(&stdout_mut);
+            pluginsd_function_result_to_stdout(transaction, HTTP_RESP_OK, "text/plain", now_realtime_sec() + 3600, wb);
+            netdata_mutex_unlock(&stdout_mut);
+            buffer_free(wb);
+            ret = HTTP_RESP_OK;
+            goto cleanup;
+        }
+        else if(!strcmp(keyword, LOGS_MANAG_FUNC_PARAM_INFO)){
+            info = true;
+        }
+        else if(!strcmp(keyword, LOGS_MANAG_FUNC_PARAM_PROGRESS)){
+            progress = true;
+        }
```

Comment:
Do we still need to keep this commented code?

Reply:
As discussed, this is WIP, so I will mark this as resolved.